### ISSUE TITLE
[discussion/feature request] Manual comment removal

### ISSUE THREAD

**Issue by AlgoRythm-Dylan** [devRant/devRant#115]
Many other social media platforms allow for users to delete comments which were made on their posts. There are many good and bad sides to implementing something like this: For example, this can really speed up the rate at which offensive content is removed as there is no need for the algo or a moderator to intervene. Trolls could be essentially eliminated (at least on others' rants). This could also work as a very caveman-style block feature where you can remove any input from people who are only there to be personally offensive.

However, of course, it is far from perfect. This can be very unattractive to users - the ability for other users to remove their freedom of speech. Also, as with any feature, it can be abused. This would put a lot of power in the hands of each ranter. I don't think the app should just give people the ability to remove others' comments, even on their own rants, and I don't think it should be a supporter feature either (It's a terrible message; "pay for your peace of mind"), but I'm sure there is some constructive effect which can come out of this train of thought, something dfox seems to agree with as well (The only reason I am writing this)

That's everything on my mind about it. Food for thought.

**Comment by DFoxinator** [comment]
Thanks for writing this up and posting it. I agree with everything you said pretty much.

This definitely has a lot of pros to it, by letting people moderate their own rants, which I've always kind of considered as belonging to them in a way. But the alternate side of that like you touched on is making it harder to have good discussions and limiting good debate. So it's tricky. 

I'm curious to hear what other people think about this and I'm hoping some more people will chime in with their thoughts.

**Comment by trogus** [comment]
I don't see devRant as a place that should be held to some lofty ideal of free speech. It's a chill place to talk about dev-related content. If you don't want to be chill or talk about dev-related content, then maybe that content doesn't belong here. So if a little self-imposed censorship is what it takes to dramatically cut down on trolls and obnoxious comments that stir shit up, I'm for it.

I'm thinking specifically of letting a rant poster block a user from further commenting (on that one rant, and potentially any of their other posted rants), and also removing undesired comments or maybe hiding/collapsing that comment in the thread, so it's still there, but only if a user goes out of their way to view it.

**Comment by AlgoRythm-Dylan** [comment]
@trogus "It's a chill place to talk about dev-related content" ignoring the part where "rant" makes up a majority of the name of your app, dev-related content often comes with many strongly held opinions. People will get angry. People will remove comments, and those who get their comments unfairly removed will retaliate. The atmosphere will become more hostile if you relax on protecting users' rights to express themselves - especially on social media. This isn't a q&a site - the point is to *talk*, and people will get unprecedentedly upset at even the slightest hindrance. Perhaps the next step - if trolls and similar people are truly a large enough concern - is to introduce a blocking feature. It's quite long overdue, and will require lots of work, but by no means will go unappreciated. That, in tandem with the report and downvote system, should protect users enough for the foreseeable future (also, your solution sounds like a smaller version of a full-blown block system. Just do the whole thing now, I vote).

A blocked user:

- Cannot comment on the rants of those whom blocked them
- Cannot see their rants in the feed
- Can still downvote their content, but they do not count (or something similar)
- Cannot reply to this user in other comment threads (Cannot tag harass)

I remember a while ago a discussion about this. Your main concern was the amount of work it would take. This is the best solution I can come up with to protect all of your lovely ranters in the least destructive way. That said, I am still just as interested and excited for additional ideas.

**Comment by trogus** [comment]
Good point, removing could definitely generate more trouble after the troll has been poked and feels the need to lash out. At the same time, the impact of this step could be powerful for reducing most users' exposure to trolling behavior (if trolling comments get hidden before most people see them and respond and further feed the troll). I wonder if other communities have tested this out or if it might be worth the experiment. Easy enough to roll-back if things go badly.

I like your ban rules, those are pretty solid.

**Comment by olegrumiancev** [comment]
Also, maybe empower the poster to moderate (either delete or collapse) the comment if it has low enough rating? <= 0 or < 0 ?

**Comment by tankerkiller125** [comment]
Maybe keep the comments and if the poster of the comment (not the ranter) gets a comment removed by the ranter then they can appeal to the moderation team in some way. That way if they are only contributing to the discussion/debate and not trolling/whatever the comment can be put back where it was.

Then if a ranter abuses their power to remove comments they should be banned from that particular feature. And all of their rants go back to the way current things are done (moderation team only)

**Comment by trogus** [comment]
Looks like Instagram is rolling out a similar feature for controlling comments on your own posts: https://techcrunch.com/2017/09/26/instagram-now-lets-you-choose-who-can-comment-on-your-posts/

Instagram has it's differences, where the poster is much more the center of attention (due to follower/following relationship) than on devRant, and so makes sense to have more control over the thread, but interesting to see if there are negative repercussions from that.
